reports/filesDataExporter/batch/batch.go

```diff
@@ -0,0 +1,179 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+
+	"github.com/iryonetwork/wwm/gen/storage/models"
+	"github.com/iryonetwork/wwm/metrics"
+	"github.com/iryonetwork/wwm/reports/filesDataExporter"
+	"github.com/iryonetwork/wwm/utils"
+)
+
+type (
+	exportError struct {
+		id  string // identifier of resource that failed to sync
+		err error
+	}
+
+	batchDataExporter struct {
+		handlers          filesDataExporter.Handlers
+		bucketsRateLimit  int
+		logger            zerolog.Logger
+		metricsCollection map[metrics.ID]prometheus.Collector
+	}
+)
+
+const exportSeconds metrics.ID = "exportSeconds"
+
+const labelFilesCollection = "filesCollection"
+
+var bucketsToSkip = map[string]bool{"encounters": true, "patients": true}
+
+func (s *batchDataExporter) Export(ctx context.Context, lastSuccessfulRun time.Time) error {
+	bucketsRateLimit := make(chan bool, s.bucketsRateLimit)
+
+	buckets, err := s.handlers.ListSourceBuckets(ctx)
+	if err != nil {
+		s.logger.Error().Err(err).Msg("failed to list source buckets")
+		return errors.Wrap(err, "failed to list source buckets")
+	}
+
+	numberOfBuckets := len(buckets)
+
+	ch := make(chan *exportError)
+	for _, b := range buckets {
+		if _, ok := bucketsToSkip[b.Name]; !ok {
```

Comment:
invalid operation: b (variable of type *invalid type) has no field or method Name